clients/src/main/java/org/apache/kafka/common/requests/EnvelopeRequest.java

```diff
@@ -91,4 +91,14 @@ public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
     public static EnvelopeRequest parse(ByteBuffer buffer, short version) {
         return new EnvelopeRequest(ApiKeys.ENVELOPE.parseRequest(version, buffer), version);
     }
+
+    public EnvelopeRequestData data() {
+        return data;
+    }
+
+    @Override
+    public Send toSend(String destination, RequestHeader header) {
+        return SendBuilder.buildRequestSend(destination, header, this.data);
```

Comment:
not sure whether it is ok to ignore the version of EnvelopeRequest.

Reply:
The version is contained in `RequestHeader`.